### ISSUE TITLE
about background task

### ISSUE THREAD

**Issue by gzxy-0102** [tiangolo/fastapi#5683]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# deploy.py
from fastapi import APIRouter, BackgroundTasks

router = APIRouter()

loaded = {}

def load_model(task_id: str):
    loaded[task_id] = "success"

@router.get("/publish")
async def publish(task_id: str, mt: str, background: BackgroundTasks):
    task = await tortoise.get(id=task_id)
    background.add_task(deploy, task.id)
    return response.success(msg="发布任务下发成功", data=task.id)

# publish.py
async def deploy(task_id:str):
    load_model(task_id)
```

### Description

I want to add data to Loaded after calling the task and get the data in the interface

But we can't get it

But the document says that if you need to access variables and objects from the same FastAPI app, or you need to perform small background tasks (like sending an email notification), you can simply just use. BackgroundTasks
![image](https://user-images.githubusercontent.com/8954610/203758207-64e1744e-a168-4ed9-ba40-266dd892af93.png)

### Operating System

Linux

### FastAPI Version

0.86.0

### Python Version

3.9

**Comment by yinziyan1206** [comment]
your code seem to have circlar imports. but it should be effective. do you use multi workers? Does your function return without any exceptions?

**Comment by gzxy-0102** [comment]
the code has no circlar imports, it's just a example, the fun has no exception

**Comment by yinziyan1206** [comment]
> the code has no circlar imports, it's just a example, the fun has no exception
> 
> > your code seem to have circlar imports. but it should be effective. do you use multi workers? Does your function return without any exceptions?

It works well when i run it. Try to debug function `deploy` and see what happens

**Comment by tiangolo** [comment]
What error are you getting? What isn't working as you would expect?

**Comment by azogue** [comment]
> I want to add data to Loaded after calling the task and get the data in the interface
> But we can't get it

@gzxy-0102, with minimal changes to make it runnable with `uvicorn` while adding some unit-testing, the equivalent code I tested is this:

```python
from uuid import uuid4

from fastapi import BackgroundTasks, FastAPI

app = FastAPI()
loaded = {}


def load_model(task_id: str):
    loaded[task_id] = "success"


async def deploy(task_id:str):
    load_model(task_id)


@app.get("/publish")
async def publish(background: BackgroundTasks):
    task_id = str(uuid4())
    background.add_task(deploy, task_id)
    return dict(data=task_id)


@app.get("/loaded")
async def get_loaded():
    return loaded


if __name__ == '__main__':
    from fastapi.testclient import TestClient

    client = TestClient(app)
    id1 = client.get("/publish").json()["data"]
    id2 = client.get("/publish").json()["data"]
    id3 = client.get("/publish").json()["data"]

    stored = client.get("/loaded").json()
    assert len(stored) == 3
    assert all(stored_id in stored.keys() for stored_id in (id1, id2, id3))
```

And works as expected, if I understood ok.

* The '/publish' endpoint initiates a background task with some ID
* The task runs and stores the state in a global variable `loaded`
* The '/loaded' endpoint access the global variable to list the IDs

As @yinziyan1206 asked,
> do you use multi workers? 

because this schema won't work in a multi-process scenario, but you're already using a database, right? (the `tortoise` code?), so you may try to store your bg-task results in the database?

**Comment by gzxy-0102** [comment]
Yes, I can also run this demo, but there are some AI related codes in it. I can only make some requests to use it in the end, but now it has no effect because I have adopted other methods
